### PR TITLE
Store invocation links in redis

### PR DESCRIPTION
**Pull request by luluz66**
Fixes https://github.com/buildbuddy-io/buildbuddy-internal/issues/1934

Also, rename some methods in ExecutionCollector interfaces to make is
clearer as it starts to collect invocation, invocation links in addition to executions.

### CODE REVIEW

File: proto/stored_invocation.proto [hunk 1]
```diff
@@ -17,3 +17,10 @@ message StoredInvocation {
   int64 invocation_status = 10;
   bool success = 11;
 }
+
+// Proto representation of the InvocationExecution. Only used in backends.
+message StoredInvocationLink {
+  string invocation_id = 1;
+  string execution_id = 2;
+  int32 type = 3;
```
Comment (vadimberezniker) [inline proto/stored_invocation.proto:25]: Can you make this an enum? It would be good if we could share the enum between this code and the mysql code.

**Review by vadimberezniker — APPROVED** [review]
I notic we don't set a TTL for the execution or the execution link keys. It would be good to do that in case the normal cleanup is not working properly for some reason. Don't need to do this in this PR though.

File: enterprise/server/backends/redis_execution_collector/redis_execution_collector.go [hunk 2]
```diff
@@ -47,6 +48,10 @@ func getInvocationKey(iid string) string {
 	return strings.Join([]string{redisInvocationKeyPrefix, iid}, "/")
 }
 
+func getInvocationLinkKey(execution_id string) string {
```
[the remark this answers is not in the document]
Comment (luluz66) [reply enterprise/server/backends/redis_execution_collector/redis_execution_collector.go]: oops. Done

File: proto/stored_invocation.proto (hunk 1 above)
Comment (luluz66) [reply proto/stored_invocation.proto]: Done

**Comment by luluz66** [comment]
> I notic we don't set a TTL for the execution or the execution link keys. It would be good to do that in case the normal cleanup is not working properly for some reason. Don't need to do this in this PR though.

I will address it in a follow-up PR.

**Pull request merged**